tests/JobCreatorTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+use PHPUnit\Framework\TestCase;
+
+class JobCreatorTest extends TestCase
+{
+    /**
+     * @dataProvider provideGetInstallerVersion
+     */
+    public function testGetInstallerVersion(
+        string $githubRepository,
+        string $branch,
+        string $expected
+    ): void {
+        $creator = new JobCreator();
+        $actual = $creator->getInstallerVersion($githubRepository, $branch);
+        $this->assertSame($expected, $actual);
+    }
+
+    public function provideGetInstallerVersion(): array
+    {
+        return [
+            // no-installer repo
+            ['myaccount/recipe-cms', '4', ''],
+            ['myaccount/recipe-cms', '4.10', ''],
+            ['myaccount/recipe-cms', 'burger', ''],
+            ['myaccount/recipe-cms', 'pulls/4/myfeature', ''],
+            ['myaccount/recipe-cms', 'pulls/4.10/myfeature', ''],
+            ['myaccount/recipe-cms', 'pulls/burger/myfeature', ''],
+            // lockstepped repo with 4.* naming
+            ['myaccount/silverstripe-framework', '4', '4.x-dev'],
+            ['myaccount/silverstripe-framework', '4.10', '4.10.x-dev'],
+            ['myaccount/silverstripe-framework', 'burger', '4.11.x-dev'],
```

Comment:
We'll have to update this test (and a few others) every time we do a minor release... can this use some smarts to grab the most recent minor here instead?

Reply:
Updated